### DISCUSSION TITLE
Specific values of parameters used by transforms

### DISCUSSION THREAD

**Discussion by mleousi** [Project-MONAI/MONAI#1392] [Q&A]
Hi all,

I have a question regarding the transforms that can be applied to the data.

For instance, when using the _RandAffined_ transform, we are giving a value to be used for the _translate_range_ and this value will then  _"be used to generate the shift parameter from uniform[-translate_range[0], translate_range[0])"_ according to the documentation. Hence, we are aware of the range from which the translate parameter is chosen, but the exact value that is finally used is not known.

I was wondering if there is a way to find out this value.

**Comment by Nic-Ma** [comment]
Hi @mleousi ,

Thanks for your interest here.
Although it's not very straightforward and not recommended, you can get the value from:
```py
trans = RandAffined()
v = trans.rand_affine.rand_affine_grid.rotate_params
```

Thanks.

**Reply by mleousi** [reply]
Hi,

Thank you for your answer.

I do have one more question on this. What happens if I use a _Compose_ block? Am I able to do something similar for each of the transforms used inside the block (supposed I have _RandAffined_ and  _Rand3DElasticd_ inside the block)?

I assume the exact values of the different transforms are more difficult to be found in this case.

Thanks.

**Reply by Nic-Ma** [reply]
Hi @mleousi ,

`Compose` saves all the transforms in a list and call them one by one when called:
https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/compose.py#L211

Thanks.
